client/post-editor/media-modal/index.jsx

```diff
@@ -129,10 +129,11 @@ export class EditorMediaModal extends Component {
 	}
 
 	componentWillMount() {
-		const { view, mediaLibrarySelectedItems, site } = this.props;
-		if ( ! isEmpty( mediaLibrarySelectedItems ) && view === ModalViews.LIST ) {
+		const { view, mediaLibrarySelectedItems, site, single } = this.props;
+		if ( ! isEmpty( mediaLibrarySelectedItems ) && ( view === ModalViews.LIST || single ) ) {
```

Comment:
In case of `single`, do you think it would make sense to avoid resetting `mediaLibrarySelectedItems` if it contains only one item?